standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -9322,9 +9334,25 @@ public void processContext(ServerContext serverContext, TTransport tTransport, T
     if (startLock != null) {
       signalOtherThreadsToStart(tServer, startLock, startCondition, startedServing);
     }
+
+    // If dynamic service discovery through ZooKeeper is enabled, add this server to the ZooKeeper.
+    if (MetastoreConf.getVar(conf, ConfVars.THRIFT_SERVICE_DISCOVERY_MODE)
```

Comment:
the variable can be METASTORE_DYNAMIC_DISCOVERY_MODE (true/false) to be inline how hiveserver has it, else atleast be METASTORE_DISCOVERY_MODE ( zookeeper)

Reply:
For some reason I do not know the names of the conf variables in MetastoreConf start with THRIFT_. There are very few variables which start with METASTORE_. I am more inclined to use METASTORE_DISCOVERY_MODE instead of THRIFT_SERVICE_DISCOVERY_MODE, but that will look odd when placed among other configurations.

It's better to use "zookeeper" instead of "true/false" since we might support more service discovery mechanisms in future.